File: src/models/VendorNotification.ts
import type { VendorNotificationStatus } from "@/types/auth";

export type VendorNotificationRow = {
  id: string;
  vendor_id: string;
  request_id: string;
  status: VendorNotificationStatus;
  quote_amount: number | null;
  quote_notes: string | null;
  quote_file_urls: string[];
  created_at: string;
};

export type VendorNotification = {
  id: string;
  vendorId: string;
  requestId: string;
  status: VendorNotificationStatus;
  quoteAmount: number | null;
  quoteNotes: string | null;
  quoteFileUrls: string[];
  createdAt: string;
};

File: rkispro-android/app/src/main/java/com/rkispro/marketplace/data/model/VendorNotification.kt
package com.rkispro.marketplace.data.model

import com.google.gson.annotations.SerializedName

data class VendorNotification(
    val id: String,
    @SerializedName("vendor_id") val vendorId: String,
    @SerializedName("request_id") val requestId: String,
    val status: String = "Sent",
    @SerializedName("quote_amount") val quoteAmount: Double? = null,
    @SerializedName("quote_notes") val quoteNotes: String? = null,
    @SerializedName("quote_file_urls") val quoteFileUrls: List<String> = emptyList(),
    @SerializedName("created_at") val createdAt: String? = null,
    @SerializedName("marketplace_request") val marketplaceRequest: MarketplaceRequest? = null
)
